### PR TITLE
Allows boiler left click to bombard again.

### PR DESCRIPTION
**Pull request by DominikPanic**
Not a bug report yet but someone reported it in-game.

### CODE REVIEW

File: code/_onclick/xeno.dm [hunk 1]
```diff
@@ -59,4 +59,12 @@
...
 	if(!is_charging)
 		return
-	stop_momentum(charge_dir)
\ No newline at end of file
+	stop_momentum(charge_dir)
+
+
+/mob/living/carbon/xenomorph/boiler/ClickOn(atom/A, params)
+	. = ..()
+	if(!selected_ability || !istype(selected_ability, /datum/action/xeno_action/activable/bombard))
```
Comment (spookydonut) [inline code/_onclick/xeno.dm]: noooooooooooooooooooooooooooo.
use a signal

Comment (DominikPanic) [reply code/_onclick/xeno.dm]: Could you give me an example? This QoL was removed in the ability refactor, and this was my best attempt to add it back.

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 2]
```diff
@@ -97,12 +97,23 @@
...
 
 /datum/action/xeno_action/activable/bombard/on_deactivation()
 	var/mob/living/carbon/xenomorph/boiler/X = owner
 	if(X.selected_ability == src)
 		X.reset_bombard_pointer()
 		to_chat(X, "<span class='notice'>You relax your stance.</span>")
+	UnregisterSignal(X, COMSIG_MOB_ATTACK_RANGED)
+
+
+/datum/action/xeno_action/activable/bombard/proc/signal_use(mob/living/carbon/xenomorph/X, atom/A, params)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm]:
```suggestion
/datum/action/xeno_action/activable/bombard/proc/on_ranged_attack(mob/living/carbon/xenomorph/X, atom/A, params)
```
A little more descriptive. Will need to change the RegisterSignal as well, of course.

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 3]
```diff
@@ -97,12 +97,23 @@
...
 	if(X.selected_ability == src)
 		X.reset_bombard_pointer()
 		to_chat(X, "<span class='notice'>You relax your stance.</span>")
+	UnregisterSignal(X, COMSIG_MOB_ATTACK_RANGED)
+
+
+/datum/action/xeno_action/activable/bombard/proc/on_ranged_attack(mob/living/carbon/xenomorph/X, atom/A, params)
+    if(!X.selected_ability)
+        return
+    if(X.selected_ability.can_use_ability(A))
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm]: this is just the same shitcode

Comment (Rohesie) [reply code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm]: LaK has asked for an example.
If you want to review PRs and for your opinion to be taken into account, then start providing some feedback.

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 4]
```diff
@@ -97,12 +97,23 @@
...
 /datum/action/xeno_action/activable/bombard/on_deactivation()
 	var/mob/living/carbon/xenomorph/boiler/X = owner
 	if(X.selected_ability == src)
 		X.reset_bombard_pointer()
 		to_chat(X, "<span class='notice'>You relax your stance.</span>")
+	UnregisterSignal(X, COMSIG_MOB_ATTACK_RANGED)
+
+
+/datum/action/xeno_action/activable/bombard/proc/on_ranged_attack(mob/living/carbon/xenomorph/X, atom/A, params)
+    if(!X.selected_ability)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm]: delete this

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm (hunk 3 above)
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm]:
```suggestion
    if(can_use_ability(A))
```

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 5]
```diff
@@ -97,12 +97,23 @@
...
+        return
+    if(X.selected_ability.can_use_ability(A))
+        X.selected_ability.use_ability(A)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm]:
```suggestion
        use_ability(A)
```

**Pull request merged**